src/mempool.rs

```diff
@@ -184,6 +179,28 @@ impl Mempool {
             let scripthash = ScriptHash::new(&txo.script_pubkey);
             self.by_funding.remove(&(scripthash, txid)); // may have misses
         }
+
+        self.modify_fee_histogram(entry.fee, -(entry.vsize as i64));
+    }
+
+    /// Apply a change to the fee histogram. Used when transactions are added or
+    /// removed from the mempool. If `vsize_change` is positive, we increase
+    /// the histogram vsize and TX count in the appropriate bin. If negative,
+    /// we decrease them.
+    fn modify_fee_histogram(&mut self, fee: Amount, vsize_change: i64) {
+        let vsize = vsize_change.unsigned_abs();
+        let bin_index = FeeHistogram::bin_index(fee, vsize);
+        let limit = 1u128 << (FeeHistogram::BINS - bin_index - 1);
+        let label = format!("[{:20.0}, {:20.0})", limit / 2, limit);
+
+        if vsize_change >= 0 {
+            self.fees.insert(bin_index, vsize);
+        } else {
+            self.fees.remove(bin_index, vsize);
+        }
+
+        self.vsize.set(&label, self.fees.vsize[bin_index] as f64);
```

Comment:
Can we skip label generation & Prometheus metrics' update during the initial sync?

Reply:
Absolutely. Actually now that I think about it, it doesn't make sense to update these labels for every single transaction. I'll push a fix which updates the metrics gauges only at the end of a mempool sync.

Reply:
Done in https://github.com/romanz/electrs/pull/970/commits/ed42680992994488961bbf5aa3028a7a3d4b5dcc